### ISSUE TITLE
Add styling option for "dirty" marker in tabs

### ISSUE THREAD

**Issue by hoschi** [onivim/oni#1580]
After @CrossR fixed #977 it would be great to give that another color and I can't find one in https://github.com/onivim/oni/blob/master/extensions/theme-onedark/colors/onedark.json

With solarized theme it is hard to see:
![2018-02-18-173636_121x28_scrot](https://user-images.githubusercontent.com/163128/36354279-463287ca-14d2-11e8-846f-669349272573.png)

**Comment by divayprakash** [comment]
@hoschi I'd like to take this up. Could you give me some more information on how to proceed?

**Comment by Akin909** [comment]
@divayprakash that'd be awesome if you wanted to take this on we'd really appreciate it. [Here](https://github.com/onivim/oni/blob/master/browser/src/UI/components/Tabs.tsx#L184) is the relevant link to the actual react component *I think (95% certain)*, at the moment its a normal react component but I think re how to implement this steps would be

1. In `Tabs.tsx` I would create a `styled-component` as we do [here](https://github.com/onivim/oni/blob/8b8a5257a856ac2e1ec0a807103c287a3e3093ba/browser/src/Services/Language/RenameView.tsx#L19) and using the theme as in the link, experiment with one of the available theme options maybe the `editor.highlight.mode.background`.
2. Bonus points: If you're interested would be creating user specifiable option, if you're interested in trying this let me know as that would be a bit more involved

**Comment by singhpratik17** [comment]
Hey, I guess this hasn't been resolved yet. If it isn't I would like to take this up as my first contribution. Does that work?

**Comment by Akin909** [comment]
@singhpratik17 that would be amazing 👍 lemme know if you need anymore details about where to go with this

**Comment by wheyerstrass** [comment]
I'm playing around with this.  
I made a `style-component` like this:  
```typescript
const DirtyMarker = styled.div`
  background-color: ${props => props.theme["foreground"]} !important;
`
```
I then replaced the current dirty marker, which is:  
```html
<div class="circle" />
```
( `circle` sets `background-color: #c8c8c8;` )
with  
```html
<DirtyMarker className="circle" />
```
Here are some examples:  

![2018-07-01-015311_124x43_scrot](https://user-images.githubusercontent.com/37278089/42129909-71b23742-7cd3-11e8-8e49-12121095b154.png)
![2018-07-01-015134_133x41_scrot](https://user-images.githubusercontent.com/37278089/42129910-71cba7fe-7cd3-11e8-94ac-fb538666ca1f.png)
![2018-07-01-014912_118x42_scrot](https://user-images.githubusercontent.com/37278089/42129911-71e46898-7cd3-11e8-88b3-e51a53bf6bfc.png)

@hoschi Look at the available colors and give me your feedback
https://github.com/onivim/oni/blob/master/extensions/theme-solarized/colors/solarized8_light.json

Having the user select the color could be done by extending the config reference e.g.  
`"ui.editor.noidea.dirtymarker.color": "#2341512"` (or use the color-keys directly), but I find this weird as it might work for one specific colorscheme but not for others.  
This seems like pure color fix due to `cirlce` class being fixed.

@Akin909 I'd appreciate your feedback.

**Comment by Akin909** [comment]
@wheyerstrass very cool thanks for having a look at that, re. the circle class if it isn't being used anywhere else I would just take the styles out of that and add it to your styled component we are slowly trying to move over all our less to styled components so that will help.

So rather than doing `!important` you could just set the color to whatever you choose and do away with the extra circle classes color (I personally like the green but it might have to come out of the theme so maybe set it to the normal highlight color then unless the user specifies a theme color it defaults to that.

For this you'd have to and add the dirty marker color option to the `Default configuration.ts`, 
 and `ConfigurationValues.ts` which is where the types are defined so in the component it would look like

```ts
	const DirtyMarker = styled<{ userColor?: string}, "div">("div")`
		//circle styles
		background-color: ${p => p.userColor || p.theme.normal.foreground(or whaterver)};
	`
// NB: the userColor would be a prop you passed down e.g.
const userColor = configuration.getValue("oni.tabs.dirtyMarker.color")
<DirtyMarker userColor={userColor} />
```

**Comment by hoschi** [comment]
@wheyerstrass nice job! I like the greenish version, should be `#2aa198` :grinning:

**Comment by hoschi** [comment]
@wheyerstrass @Akin909 good points both of you. I find it important to see modification color, because of that I would like the green more than the black (foreground) color. Which answers the "should it be changeable" question :grin: It should be as different people focus/like different things/styles. My suggestions:
* make it changeable
* use foreground color as default, because for most themes this should be outstanding enough

Deal?

**Comment by wheyerstrass** [comment]
Jep, I'm cool with either - just getting the hang of everything.

**Comment by wheyerstrass** [comment]
I created a pull request: #2391 
Tell me what to improve ( I'm currently thinking about tests ).

**Comment by Akin909** [comment]
I think this can now be closed do you agree @hoschi?

**Comment by hoschi** [comment]
Yes

Akin <notifications@github.com> schrieb am Mo., 16. Juli 2018, 11:46:

**Comment by Akin909** [comment]
closed by #2397
